app/controllers/users/passwords_controller.rb

```diff
@@ -6,12 +6,27 @@ class Users::PasswordsController < Devise::PasswordsController
   layout "session"
 
   def create
-    if User.find_for_authentication(resource_params.permit(:login))&.prevent_password_resets?
+    user = User.find_for_authentication(resource_params.permit(:login))
+
+    if user&.protected_user || user&.prevent_password_resets?
       flash[:error] = t(".reset_blocked", contact_abuse_link: view_context.link_to(t(".contact_abuse"), new_abuse_report_path)).html_safe
       redirect_to root_path and return
+    elsif user&.password_resets_limit_reached?
+      available_time = ApplicationController.helpers.time_in_zone(
+        user.password_resets_available_time, nil, user
+      )
+
+      flash[:error] = t(".reset_cooldown", reset_available_time: available_time).html_safe
+      redirect_to root_path and return
     end
+
     super do |user|
-      flash.now[:notice] = ts("We couldn't find an account with that email address or username. Please try again?") if user.nil? || user.new_record?
+      if user.nil? || user.new_record?
+        flash.now[:notice] = ts("We couldn't find an account with that email address or username. Please try again?")
+      else
+        user.update_password_resets_requested
+        user.save
```

Comment:
Hmm, I'm not sure if this should be `user.save` or `user.save!`. On the one hand, it could be a good idea to make sure the reset counter is incremented before sending out the email, but on the other it seems a bit draconian to not send the email if something goes wrong saving... @zz9pzza (if you have time) what do you think?